awaitwhat/blocker.py

```diff
@@ -12,9 +13,13 @@ def blockers(task):
 
     stack = task_get_stack(task, None)
 
-    if len(stack) > 2 and sleep.mine(stack[-2]):
-        return [sleep.decode(stack[-2])]
-
+    if len(stack) > 2:
+        if sleep.mine(stack[-2]):
+            return [sleep.decode(stack[-2])]
+        elif wait_for.mine(stack[-2]):
+            wait_for_string, awaitable = wait_for.decode(stack[-2])
+            return [wait_for_string, awaitable]
```

Comment:
I'm not sure about changing the signature of `decode` methods... Let me poke a bit at that.

Reply:
In the end I've relaxed the signature a little bit.
strings were (apparently) always valid in the output of `decode`.

Reply:
I tried my best to keep the signature the same. But since wait for does not expose(returns) the awaitable required, i was forced to return it. However, this is an excellent refactor!